src/modules/displaymanager/main.py

```diff
@@ -439,22 +442,30 @@ def run():
 
             if default_desktop_environment is not None:
                 os.system(
-                    "sed -i -e \"s/^.*user-session=.*/user-session={!s}/\" "
-                    "{!s}/etc/lightdm/lightdm.conf".format(
+                    "sed -i -e \"s/^.*user-session=.*/user-session={!s}/\" " "{!s}".format(
                         default_desktop_environment.desktop_file,
-                        root_mount_point
-                        )
-                    )
-
-            if default_desktop_environment.desktop_file == "deepin":
-                os.system(
-                    "sed -i -e \"s/^.greeter-session=.*"
-                    "/greeter-session=lightdm-deepin-greeter/\" "
-                    "{!s}/etc/lightdm/lightdm.conf".format(
-                        root_mount_point
+                        lightdm_conf_path
                         )
                     )
 
+            # configure lightdm-greeter
+            greeter_path = os.path.join(
+                root_mount_point, "usr/share/xgreeters"
+                )
+            if (os.path.exists(greeter_path)):
+                greeter = os.listdir(greeter_path)[0].split('.')[0]
+                libcalamares.utils.debug("configure {!s}".format(greeter))
+                with open(lightdm_conf_path, 'r') as lightdm_conf:
+                    text = lightdm_conf.readlines()
+
+                with open(lightdm_conf_path, 'w') as lightdm_conf:
```

Comment:
Here you're doing sed -i in Python; it feels weird (but not wrong) to have two really different ways of replacing a line in a text file, within one Python module.